packages/alchemy/src/client/rpcClient.ts

```diff
@@ -25,6 +26,10 @@ export const createAlchemyPublicRpcClient = ({
       ? `${chain.rpcUrls.alchemy.http[0]}/${connectionConfig.apiKey ?? ""}`
       : connectionConfig.rpcUrl;
 
+  if (!rpcUrl) {
+    throw new InvalidRpcUrlError({ chain, connectionConfig });
+  }
```

Comment:
rpcUrl can only be falsy here if `connectionConfig.rpcUrl` is `""` right? why not just do something like this above this if-statement: 

```
const rpcUrl =
    connectionConfig.rpcUrl ?? `${chain.rpcUrls.alchemy.http[0]}/${connectionConfig.apiKey ?? ""}`;
```

Reply:
actually I think I need to update the above some more... this is meant to catch the bad chain config